### PR TITLE
Regions tests

### PR DESCRIPTION
**Pull request by AlexKaschuk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+            _regionService.Setup(x => x.GetAllRegionsAsync()).ReturnsAsync(GetRegions());
+            // Act
+            var result = await _RegionController.Index();
+            // Assert
+            Assert.IsInstanceOf<OkObjectResult>(result);
+            Assert.IsInstanceOf<List<RegionDTO>>((result as ObjectResult).Value);
+        }
+
+
+
```
Comment (tetyanakutsko) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:49]: Remove unnecessary spaces

**Review by tetyanakutsko — COMMENTED** [review]
Great job !

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+            Assert.IsInstanceOf<OkResult>(result);
+        }
+
+
+
+        [Test]
+        public async Task CreateRegion_NullRegion_ReturnsOkResult()
+        {
+            // Arrange
+
```
Comment (LanchevychMaxym) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: Redundant comment and line

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+            // Assert
+            Assert.IsInstanceOf<OkResult>(result);
+        }
+
+
+        [Test]
+        public async Task GetPhotoBase64_NullString_ReturnsOkObjResult()
+        {
+            // Arrange
+
```
Comment (LanchevychMaxym) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: the same issue

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+            // Assert
+            Assert.IsInstanceOf<OkObjectResult>(result);
+            Assert.IsNull((result as ObjectResult).Value);
+        }
+
+        [Test]
+        public async Task RedirectCities_AnyInt_ReturnsOkResult()
+        {
+            // Arrange
+
```
Comment (LanchevychMaxym) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: ?

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+            // Assert
+            Assert.IsInstanceOf<OkResult>(result);
+        }
+
+
+        [Test]
+        public async Task EditRegion_NullInt_ReturnsOkResult()
+        {
+           
+
```
Comment (LanchevychMaxym) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: ?

**Review by LanchevychMaxym — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 5 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 4 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 3 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 2 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 1 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

**Review by bodnarira — APPROVED** [review]

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+using System.Threading.Tasks;
+
+namespace EPlast.Tests.Controllers
+{
+    class RegionsControllerTests
+    {
+        private  Mock<ILoggerService<CitiesController>> _logger;
+        private  Mock<IRegionService> _regionService;
+        private  Mock<IRegionAdministrationService> _regionAdministrationService;
+        private  Mock<IRegionAnnualReportService> _RegionAnnualReportService;
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:20]:
```suggestion
        private  Mock<IRegionAnnualReportService> _regionAnnualReportService;
```

File: EPlast/EPlast.Tests/Services/Regions/RegionServiceTest.cs [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+            new CityDTO { ID = 2, Name = "Перемишляни" }
+        };
+
+        private readonly IEnumerable<DataAccess.Entities.City> city = new List<DataAccess.Entities.City>
+        {
+            new DataAccess.Entities.City { ID = 1, Name = "Золочів" },
+            new DataAccess.Entities.City { ID = 2, Name = "Перемишляни" }
+        };
+
+        private readonly IEnumerable<RegionAdministration> Admins = new List<RegionAdministration>
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/Regions/RegionServiceTest.cs:386]:
```suggestion
        private readonly IEnumerable<RegionAdministration> _admins = new List<RegionAdministration>
```

**Review by voroniak — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 6 above)
Comment (ostappppppp) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:20]: as all of yours properties named form lowercase, rename _RegionAnnualReportService

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,430 @@
...
+
+        private RegionsController _RegionController;
```
Comment (ostappppppp) [inline EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs:22]: also here

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 6 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

File: EPlast/EPlast.Tests/Services/Regions/RegionServiceTest.cs (hunk 7 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Services/Regions/RegionServiceTest.cs:386]: done

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 6 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

File: EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs (hunk 8 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.Tests/Controllers/RegionsControllerTests.cs]: done

**Review by LanchevychMaxym — APPROVED** [review]

**Review by voroniak — APPROVED** [review]

**Pull request merged**